tensorflow/compiler/xla/service/gpu/target_util.h

```diff
@@ -76,8 +76,12 @@ llvm::Value* EmitCallToTargetFunction(
     absl::Span<llvm::Type* const> overloaded_types, llvm::IRBuilder<>* b);
 
 // Obtain the target specific address space for global variables
-unsigned int GetGlobalMemoryAddressSpace(llvm::Module* module);
-unsigned int GetSharedMemoryAddressSpace(llvm::Module* module);
+unsigned int GetGlobalMemoryAddressSpace(const llvm::Module& module);
+unsigned int GetSharedMemoryAddressSpace(const llvm::Module& module);
+
+// Annotate the kernel as GPU kernel according to the GPU target.
+void AnnotateKernelAsGpuKernel(llvm::Module* module, llvm::Function* kernel,
```

Comment:
@sumesh13 `AnnotateKernelAsGpuKernel` still sounds like palindrome to me. would it be more suitable if we name it `AnnotateFunctionAsGpuKernel`? From LLVM perspective the 2nd parameter is essentially just a regular LLVM function, it's only until this function is invoked would metadata or attributes of GPU kernel be applied.